getignore.go

```diff
@@ -24,12 +25,32 @@ func (fetcher *ignoreFetcher) NameToUrl(name string) string {
 	return fetcher.baseUrl + "/" + name + ".gitignore"
 }
 
-func getNames(f *os.File, namesChannel chan string) {
+func addNamesToChannel(names []string, namesChannel chan string) {
+	for _, v := range names {
+		namesChannel <- v
+	}
+	close(namesChannel)
+}
+
+func getNamesFromArguments() []string {
+	filePointer := flag.String("file", "", "Path to file of names")
+	flag.Parse()
+	names := flag.Args()
+
+	if *filePointer != "" {
+		namesFile, _ := os.Open(*filePointer)
+		names = append(names, getNamesFromFile(namesFile)...)
```

Comment:
After looking at this again, I don't know what the best solution is.  Some thoughts:

The problem is, you have `flag.Args`, and potentially the `files` flag.  One, or the other, or both.

Th idea was that the `--files` flag should be optional; I think it's probably most common that you'd want to do `./getignore foo bar baz`, with no file at all.  But the way its currently implemented, `./getignore --file=moreNames.txt foo bar` would be valid as well, as would `./getignore --file=names.txt`.  So we have a single function that 

* Parses all the args and/or flags
* Returns a single array of names.

I guess we could have one function that checks for the `file` flag, opens and returns the file handle.  But what if they didn't pass the flag?  Then grab `flag.Args`, and concat both arrays in `main`?  That doesn't seem cleaner to me, but I'm open to suggestions.

As far as the names channel:  I thought the idea was that we can process a channel in parallel?  So we (have to) load the names in sequentially, but building the URLs and making requests can all happen in parallel.  Is that wrong?